src/Ansatz.jl

```diff
@@ -308,13 +312,11 @@ end
 function truncate!(::MixedCanonical, tn::AbstractAnsatz, bond; threshold, maxdim)
     # move orthogonality center to bond
     mixed_canonize!(tn, bond)
-    return truncate!(NonCanonical(), tn, bond; threshold, maxdim, compute_local_svd=false)
+    return truncate!(NonCanonical(), tn, bond; threshold, maxdim, compute_local_svd=true)
 end
 
 function truncate!(::Canonical, tn::AbstractAnsatz, bond; threshold, maxdim)
-    truncate!(NonCanonical(), tn, bond; threshold, maxdim, compute_local_svd=false)
-    # requires a sweep to recanonize the TN
-    return canonize!(tn)
+    return truncate!(NonCanonical(), tn, bond; threshold, maxdim, compute_local_svd=false)
```

Comment:
mmm you're missing the contraction with lateral $\Lambda$ and the reconstruction no?

Reply:
I don't see why this is necessary, when you have already the things in the `Canonical` form the singular values in the bonds can already be used to `truncate` the `MPS`, since you can view like if you are in the orthogonality center.

Reply:
right, @starsfordummies ?